### ISSUE TITLE
ArrayIndexOutOfBoundsException in GridScrollExtension.getColumnWidth

### ISSUE THREAD

**Issue by alexandru-prigoreanu** [TatuLund/grid-scroll-extension#17]
hello! I encountered an ArrayIndexOutOfBoundsException using the following code
```        Window window = new Window("Window caption");
        Grid<Pair<String, String>> grid = new Grid<>();
        GridScrollExtension<Pair<String, String>> gridScrollExtension = new GridScrollExtension<>(grid);
        grid.addColumn(Pair::getLeft).setCaption("Left").setId("left");
        Button addRightColumn = new Button("Add right column",
                event -> grid.addColumn(Pair::getRight).setCaption("Right").setId("right"));
        addRightColumn.setDisableOnClick(true);
        grid.addColumnResizeListener(event -> grid.getColumns()
                .forEach(column -> log.info("the width of the column {} is {}", column.getId(),
                        gridScrollExtension.getColumnWidth(column.getId())))); //on this line here
        window.setContent(new VerticalLayout(grid, addRightColumn));
        UI.getCurrent().addWindow(window);
```
with the following user interaction:
1. a window gets shown with VerticalLayout containing a grid with one "Left" column and a button that will add a "Right" column to the grid
2. the user resizes the width of the "Left" column
3. the user clicks the button "Add right column" and the "Right" column is added to the grid
4. the user resizes the width of the "Left" column raising the following exception
```
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at org.vaadin.extension.gridscroll.GridScrollExtension.getColumnWidth(GridScrollExtension.java:275)
```
An example of the broken ui can be found in the attached screenshot.
<img width="1463" alt="github" src="https://user-images.githubusercontent.com/74921454/123726111-62700a00-d85d-11eb-8f8d-ad490d3274bb.png">

Do you have any suggestions on the matter? Am i using the api in a wrong way? If there is a bug and you pinpoint me where and how to solve it i can try to create a pull request.
thank you for your time!

**Comment by TatuLund** [comment]
There are two issues here, the first one is in your code, and there is also a bug in the add-on.

1. There is a bug in the add-on. Column widths should be reported also when `ColumnResizeCompensationMode` is `NONE`. I have fixed this for version 2.5.1.

2. There is a timing issue in Grid, the `grid.addColumnResizeListener(..)` fires too early and is processed before column widths are available. Because of this the extension itself provides `gridScrollExtension.addColumnResizedListener(..)`, which you should use instead.

**Comment by alexandru-prigoreanu** [comment]
@TatuLund thank you very much!
